hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HiveTypeUtils.java

```diff
@@ -259,22 +260,22 @@ private static TypeInfo generateTypeInfoWorker(Schema schema,
     }
   }
 
-  private static TypeInfo generateRecordTypeInfo(Schema schema,
-                                                 Set<Schema> seenSchemas) throws AvroSerdeException {
-    assert schema.getType().equals(Schema.Type.RECORD);
+  private static TypeInfo generateRecordTypeInfo(HoodieSchema schema,
+                                                 Set<HoodieSchema> seenSchemas) throws AvroSerdeException {
+    ValidationUtils.checkArgument(schema.getType() == RECORD, schema + " is not a RECORD");
```

Comment:
Use the supplier for the message to avoid eagerly generating the error message for all usages in this class